app/models/page.rb

```diff
@@ -17,6 +21,28 @@ class Page < ApplicationRecord
   mount_uploader :social_image, ProfileImageUploader
   resourcify
 
+  # @param slug [String]
+  #
+  # @return An HTML safe String.
+  #
+  # @yield Yield to the calling context if there's no Page match for slug.
+  #
+  # @note Yes, treating this value as HTML safe is risky.  But we already opened that vector by
+  #       letting the administrator of pages write HTML.
+  #
+  # @todo Do we want to only allow certain slugs?
+  #
+  # rubocop:disable Rails/OutputSafety
+  def self.render_safe_html_for(slug:)
+    page = find_by(slug: slug)
+    if page
+      page.processed_html.html_safe
+    else
+      yield
```

Comment:
ruby style question: this `yield` assumes a block was given.
Does it make sense to add &block to the parameter list (in case I'm too lazy to read your documentation comment immediately before, but smart enough to have looked at the method definition). I don't know that we do that anywhere else.
Should we have any fallback behavior if a block was not given?

Is there an advantage or future use case that passing a block here enables, or are we always going to pass a block calls render on a fixed view identifier? Could we pass a string fallback_view like "coc-text" and call `render fallback_view` directly?

Reply:
I had done some consideration regarding a fallback option.  I really didn't like any of them.  In part because the model doesn't have the view context, and I'd prefer to not pass that in.

I think we could replace `yield` with `block.call` and document that it's almost certainly expected.

I can add a test stating if we don't find a slug and don't pass a block, we raise an exception.  That seems like a clearer statement of interaction.

Reply:
>  the model doesn't have the view context

ah - you can't _just_ call `render`, we're using the block as a closure to capture the surrounding context. Thanks!